no-test-support-import false positives

C:\app\tests\test-helper.js

```js
import { setApplication } from "@ember/test-helpers";
import start from "ember-exam/test-support/start";
```

```text
C:\app\tests\test-helper.js
  2:1  error  Do not import a file from test-support into production code, only into test files  ember/no-test-support-import
```

C:\app\tests\unit\controllers\reseller\dashboard-test.js

```js
import { module, test } from "qunit";
import { setupTest } from "ember-qunit";
import window from "ember-window-mock";
import { setupWindowMock } from "ember-window-mock/test-support";
```

```text
C:\app\tests\unit\controllers\reseller\dashboard-test.js
  4:1  error  Do not import a file from test-support into production code, only into test files  ember/no-test-support-import
```

--- a/lib/rules/no-test-support-import.js
+++ b/lib/rules/no-test-support-import.js
@@ -4,16 +4,10 @@
 
 'use strict';
 
+const path = require('path');
+
 const ERROR_MESSAGE_NO_IMPORT =
   'Do not import a file from test-support into production code, only into test files.';
-
-function hasTestSupportDirectory(importSource) {
-  return (
-    importSource.startsWith('test-support/') ||
-    importSource.endsWith('/test-support') ||
-    importSource.includes('/test-support/')
-  );
-}
 
 module.exports = {
   ERROR_MESSAGE_NO_IMPORT,
@@ -32,17 +26,19 @@ module.exports = {
 
   create: function create(context) {
     const fileName = context.getFilename();
+    const fileNameParts = fileName.split(path.sep);
 
     return {
       ImportDeclaration(node) {
         const importSource = node.source.value;
+        const importSourceParts = importSource.split(path.sep);
 
         if (
-          hasTestSupportDirectory(importSource) &&
+          importSourceParts.includes('test-support') &&
           !(
-            fileName.includes('/tests/') ||
-            fileName.includes('/addon-test-support/') ||
-            fileName.includes('/test-support/')
+            fileNameParts.includes('tests') ||
+            fileNameParts.includes('addon-test-support') ||
+            fileNameParts.includes('test-support')
           )
         ) {
           context.report({

